### PR TITLE
[Draft] Add experimental support for incremental annotation processing in mini-processor

### PR DESCRIPTION
**Pull request by adriangl**
### PR's key points
The PR adds experimental support for incremental annotation processing compilation while using `mini-processor`

This is done making the annotation processor an `Aggregating` annotation processor. You can read more about this in Gradle's official documentation here: https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing.

You can see some implementations of such annotation processors in the links provided in `MiniProcessor`.

I also rewrote some code from the processors to make it a bit easier to understand

### How to review this PR?
Basically test if builds are faster with the version in this branch and the tips mentioned in the `README.md`

### CODE REVIEW

File: mini-processor/build.gradle [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 apply from: "../jitpack.gradle"
 
 dependencies {
     api project(":mini-common")
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.github.yanex:takenoko:0.1'
-    implementation 'com.squareup:kotlinpoet:1.3.0'
+    implementation 'com.squareup:kotlinpoet:1.5.0'
+
+    // Lib to add annotation processing
```
Comment (adriangl) [inline mini-processor/build.gradle]:
```suggestion
    // Lib to add incremental annotation processing
```

File: README.md [hunk 2]
```diff
@@ -202,6 +202,31 @@ dependencies {
...
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+```
+#### Improve compilation speed
+In order to speed up the compilation process, it is recommended to add the following settings in
+your `gradle.properties`:
+```groovy
+// Improves kapt speed with parallel annotation processing tasks, may impact in memory usage
```
Comment (adriangl) [inline README.md]:
```suggestion
## Improves kapt speed with parallel annotation processing tasks, may impact in memory usage
```

File: README.md [hunk 3]
```diff
@@ -202,6 +202,31 @@ dependencies {
...
+kapt.use.worker.api=true
+// Enables Gradle build cache
```
Comment (adriangl) [inline README.md]:
```suggestion
## Enables Gradle build cache
```
